src/Url.php

```diff
@@ -1066,4 +1066,35 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $parsedUrl = parse_url($url);
+
+    // Test parsed URL
+    foreach ($candidates as $candidate) {
+      if(preg_match('/' . $candidate . '\.(htm(l?)|php/', $parsedUrl["path"])) {
+        $root_path = substr($parsedUrl, strlen($parsedUrl) - strlen($candidate));
+        break;
+      }
+    }
+
+    return $root_path;
```

Comment:
That is the intended case... if we have a path, does it reduce to some default path as well.

Examples: 
section/index.html  reduces to section/ so that needs to be returned.
section/index.html?page=boo  reduces to section?page=boo so that has to be returned.